akka-http-core/src/test/scala/akka/http/engine/ws/FramingSpec.scala

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.engine.ws
+
+import scala.collection.immutable
+import scala.concurrent.duration._
+
+import org.scalatest.matchers.Matcher
+import org.scalatest.{ FreeSpec, Matchers }
+
+import akka.util.ByteString
+
+import akka.stream.scaladsl.Source
+import akka.stream.stage.Stage
+
+import akka.http.util._
+
+import Protocol.Opcode
+
+class FramingSpec extends FreeSpec with Matchers with WithMaterializerSpec {
+  import BitBuilder._
+
+  "The Websocket parser/renderer round-trip should work for" - {
+    "the frame header" - {
+      "interpret flags correctly" - {
+        "FIN" in {
+          b"""1000     # flags
+                  0000 # opcode
+              0        # mask?
+               0000000 # length
```

Comment:
Yes. I think it fits in the testkit. Useful when writing/testing stages/bidiflows which work with binary.